### DISCUSSION TITLE
Messages stuck in queue

### DISCUSSION THREAD

**Discussion by brywhi** [postalserver/postal#1778] [Help with using Postal] [answered]
I'm having a issue with mails getting stuck in queue, and I'm not sure why. I had about 200-300 this morning. We have very low volume, usually only 2-10 messages per minute. I had about 225 messages stuck in queue for over 24 hours. I got about 160 of them to send by doing the following, but I can't get the rest to send. What should I check to figure out why?

`Postal status ` shows that all containers are up.

Stopped postal:

`sudo postal stop`

Entered the mysql docker container:

`docker exec -it postal-mariadb bash`

Then I ran mysql -u root -p and looked at all the locked messages. They had all been locked for over 24 hours with 0 attempts.

`use postal; select * from queued_messages;
`
Then I unlocked them all

`use postal; update queued_messages set locked_by = NULL, locked_at = NULL, retry_after = NULL, ip_address_id = NULL;
`
Started postal:

`sudo postal start`

**Comment by brywhi** [comment, marked as answer]
Actually, nevermind- I found the issue. It seems that my server was running low on ram. This makes sense, because I just increased the ram limit on maraidb so that I could send larger emails- which constrained the ram on our cheap VPS. I noticed because I finally saw an error at the right time and the worker container failed with status 137- out of memory- when it got to a larger message in the queue to send.

To work around this issue, since we have very low volume, I just added a swap file to our host so ram can overflow there if ever needed. The remaining queue emptied itself out, and now it seems to be fine.

**Reply by willpower232** [reply]
Thanks for the detail, memory issues can be hard to spot.
